src/utils/rooms.ts

```diff
@@ -14,7 +14,19 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+import { IInstance, IProtocol, IPublicRoomsChunkRoom, MatrixClient } from "matrix-js-sdk/src/client";
+
+import { Action } from "../dispatcher/actions";
+import { ViewRoomPayload } from "../dispatcher/payloads/ViewRoomPayload";
 import { getE2EEWellKnown } from "./WellKnownUtils";
+import dis from "../dispatcher/dispatcher";
+import { getDisplayAliasForAliasSet } from "../Rooms";
+import { _t } from "../languageHandler";
+import { ALL_ROOMS, Protocols } from "../components/views/directory/NetworkDropdown";
+import { instanceForInstanceId, protocolNameForInstanceId } from "./DirectoryUtils";
+import SdkConfig from "../SdkConfig";
+import Modal from "../Modal";
+import ErrorDialog from "../components/views/dialogs/ErrorDialog";
```

Comment:
Importing components in utilities is reserved for utilities ending in `.tsx`, which must be separate from other utilities for import cycling reasons down the line.